### ISSUE TITLE
vim-slime should ask for a socket path, not a socket name

### ISSUE THREAD

**Issue by thejohnfreeman** [jpalardy/vim-slime#99]
tmux supports two ways to pick the session's socket from the command line:

`-L` : User passes a name of a socket that exists in `$TMPDIR`
`-S` : User passes an absolute path to a socket

On shared hosts, it is not safe to have tmux putting its sockets into the default value for `$TMPDIR`, `/tmp`, because they are likely to be blown away. However, it is often undesirable to set `$TMPDIR` to a path in the home directory because other processes can fill it up with garbage. tmux switched to a separate variable, `$TMUX_TMPDIR`, but not until recently, some time after the latest version in use at my company (1.8). Altogether, it makes the use of `-L` very hard to get right. For example, vim-slime simply cannot function when I use 1.8, keep my tmux socket in my home directory on a shared machine, but leave `$TMPDIR` unset or pointing at `/tmp`.

To get around this issue, I think it would be better if vim-slime asked for a full path to the socket and used `-S` internally. This will require plugin users to type more than a socket name, but I imagine the vast majority of users are sending snippets to the same session, not a different session. The current session's socket path is kept in `$TMUX` and can be parsed in vim with `split($TMUX, ',')[0]`. It can be used as the default path and fit the needs of most users [^1]. I have made [this improvement before for vim-tmux-navigator](https://github.com/christoomey/vim-tmux-navigator/pull/88/commits/5d76881c6ee5de89994ef7e2308bb1ddaef38bea). I will be happy to volunteer a patch this evening, but it will be your decision whether to accept it.

[^1]: I honestly suspect it will fit the need of every user, and thus the prompt can be removed entirely, but I might be wrong.

**Comment by jpalardy** [comment]
I commented on the PR.

I'm not quick sure what the right solution is…

**Comment by pinggit** [comment]
so how to make it work with tmux1.8? I am in an environment that I have no privilege to upgrade tmux, and found vim-slime does not work...

**Comment by jpalardy** [comment]
Hi @pinggit 

Please open another issue and describe your situation. We'll see what we can do
